### ISSUE TITLE
[BUG] DNS Issue with quote marks being added

### ISSUE THREAD

**Issue by ledoktre** [hestiacp/hestiacp#2377]
I have noticed for a while now that when you add a TXT DNS record, quotes are getting added automatically around the IP or Value column. They show up in the record list, and when you edit the record.  

If you edit the record, remove the quotes, and hit save, they stay gone.  I believe they should not be added in the first place unless someone explicitly does so on their own.

This appears to take place with latest version of Hestia (and all versions since I started using the panel a year or more ago).  It happened with latest versions of Firefox, Chrome, and Brave browsers- thats all I tested with.

Thanks,

**Comment by jaapmarcus** [comment]
They should only be added when you enter an   in the value field....

**Comment by ledoktre** [comment]
Agreed.  When I typically see this is when copying and pasting SPF records or DKIM records.  They don't have quotes around them on the source material or the input field on Hestia.  When I save it and bring up the records list, they are there.  If I edit the record, they are there (added).  If I delete them from the beginning and end and hit save, they are gone.

Not sure why they get added either.  They show up with manually typed TXT records as well so its not a copy/paste thing that I can tell.

Thanks

**Comment by ledoktre** [comment]
Here is a link to a quick screen cast showing the issue:

https://sync.ledoktre.com/f/cdccbd4b69a342839d0d/

**Comment by ledoktre** [comment]
Here is a link to a quick screen cast showing the issue:

https://sync.ledoktre.com/f/cdccbd4b69a342839d0d/

**Comment by ledoktre** [comment]
It seems to be things with spaces in them triggers it. I added another TXT record without spaces, and no quotes were added.

**Comment by jaapmarcus** [comment]
They should be added if the the string contains a space otherwise the .db will not load correctly so the behaviour is correct the only bug is then when saving we should check if there spaces present and add them if needed

**Comment by ledoktre** [comment]
Gotcha. My concern was quotes being added to the actual bind records that would cause issues with DNS lookups.  Possibility of double quotes if something contained quotes for whatever reason when copied in, etc. Thanks

**Comment by jaapmarcus** [comment]
We should also probally check if they have allready added.. But I think it is allready done ...

https://github.com/hestiacp/hestiacp/blob/25c7553a83461617981ef5ac22725483c0e5ef33/bin/v-add-dns-record#L68-L74

**Comment by ledoktre** [comment]
I suppose also if the quotes are there by design then, then the real bug might also be when you edit the record, remove and save, they should be added back in (maybe thats what you meant above, not sure. Been a long day!)

**Comment by jaapmarcus** [comment]
Yes that is correct ..

**Comment by ledoktre** [comment]
It might not be a feasible thing to do, but if this tripped me up, it may trip up others.  Is there such a thing as the opening and closing quotes being hidden from the zone list and the create/edit screens?  Sort of a behind the scenes transparent thing.  I was getting caught up in some having quotes, some not having quotes. Other panels not having quotes in some cases for SPF records, etc.

**Comment by jaapmarcus** [comment]
I think it mainly applies for text records.. 

There might also be a bug with very long DKIM keys.

**Comment by jaapmarcus** [comment]
Did some testing:

```
root@dev:~# v-change-dns-record test abc.com 13 'test' TXT 'test kees'
root@dev:~# v-list-dns-records test abc.com
```
And then echo the .db file:

test	14400	IN	TXT		"test kees"

Even if you remove the " it will get added in the .db file

Question to be asked that behaviour do we want

**Comment by ledoktre** [comment]
If I am understanding your last comment properly, you are saying that testing with and without " marks the " mark gets added to the .db file just fine.

If thats the case, I am in favor of losing the " marks from the interface.  Personally I think whatever gets entered in should be what gets (perceived anyway) saved, as long as syntax checks pass.

My confusion was because I didn't enter quotes yet they showed up in the web UI after saving. And compounded when I deleted them, saved, and they appeared to stay gone. If we are handling the " behind the scenes as it appears is the case, I'd say the logic for creating the TXT record should be more like the update logic for TXT records.  Update works as desired/expected IMHO.

That is of course my opinion, for whatever it may be worth :-)
